### PR TITLE
docs: specs for progressive mp, default, ephemeral spaces

### PR DESCRIPTION
**Pull request by zhenyasav**
### Summary
🛠️🌟🗑️

This pull request updates the docs and specs for the DXOS framework, mainly focusing on the new ECHO API and the DXOS Shell. It removes outdated and experimental files, renames files to match the current version, and adds new files to explain new concepts and features. It also updates the code examples and the QA script to reflect the latest changes and scenarios.

> _We create and destroy the spaces of ECHO_
> _We persist and replicate our state across the void_
> _We defy the identity choices that bind us_
> _We unleash the power of the DXOS Shell_

### Walkthrough
*  Simplify the API for creating spaces without the `echo` namespace ([link](https://github.com/dxos/dxos/pull/3814/files?diff=unified&w=0#diff-940d4ad80d17cda879628545b858e377523ba0f31d3e69b8b963b591e4ab2c7fL13-R13))
*  Introduce the concept of ephemeral spaces for storing ephemeral runtime state ([link](https://github.com/dxos/dxos/pull/3814/files?diff=unified&w=0#diff-883fbfe17a805f9b90f03251891dbd8b52b08dfb7aca971ad1fa08e4b17afdfdR1-R61))
*  Introduce the concept of personal spaces for persisting state across devices ([link](https://github.com/dxos/dxos/pull/3814/files?diff=unified&w=0#diff-ccc75917922fb836b5835925642b6e521141f7e4838caeded46e95d3fdd97f5eR1-R61))
*  Introduce the API for ECHO that supports progressive multiplayer without identity choices ([link](https://github.com/dxos/dxos/pull/3814/files?diff=unified&w=0#diff-355aa2477564860cd74a882226517cf68b7837c46f88b0d5807d5e5853791251L1-R89))
*  Introduce the DXOS Shell component for guiding users through identity and space operations ([link](https://github.com/dxos/dxos/pull/3814/files?diff=unified&w=0#diff-828a51109970a741cb95d05117de0dc8c30177af007c33f2e193656a30586f06R1-R13))
*  Update the QA script to reflect the current features and scenarios ([link](https://github.com/dxos/dxos/pull/3814/files?diff=unified&w=0#diff-22fc15cdef2bc7e2a38cb8d0386142c4866df516b703555a039797412ccddc42R1-R38))
*  Remove outdated and experimental code from the `qa.md` file and move it to `experimental/qa.md` ([link](https://github.com/dxos/dxos/pull/3814/files?diff=unified&w=0#diff-6aee7b1d212108599b6e43962b13d7888d3a0c7cd27f4b034477814df3987a75L1-L19), [link](https://github.com/dxos/dxos/pull/3814/files?diff=unified&w=0#diff-6aee7b1d212108599b6e43962b13d7888d3a0c7cd27f4b034477814df3987a75L44-L56), [link](https://github.com/dxos/dxos/pull/3814/files?diff=unified&w=0#diff-6aee7b1d212108599b6e43962b13d7888d3a0c7cd27f4b034477814df3987a75L117), [link](https://github.com/dxos/dxos/pull/3814/files?diff=unified&w=0#diff-6aee7b1d212108599b6e43962b13d7888d3a0c7cd27f4b034477814df3987a75L134-L141))
*  Rename the files in the `specs/v0.2` folder to `specs/v0.1` to avoid confusion ([link](https://github.com/dxos/dxos/pull/3814/files?diff=unified&w=0#diff-54045fb0382975e84e5c591a9c11ad3e444b32ced2cc4155b3ac9d696bb074f6), [link](https://github.com/dxos/dxos/pull/3814/files?diff=unified&w=0#diff-e6c1378e2dda08da35f20224c3474436a906d78ff265e83cc36ec5967f5aa418), [link](https://github.com/dxos/dxos/pull/3814/files?diff=unified&w=0#diff-35637ee4803ebecf53deceb23f8f16f0f057289565e7e122e279c60d21b4b95a))
*  Delete the duplicate `shell.md` file in the `specs/v0.2` folder ([link](https://github.com/dxos/dxos/pull/3814/files?diff=unified&w=0#diff-70279cb2ad1ad71070cb24ba3f26ab5e2bdd16cd63005e6cf4bb9c726d2c5e0f))

### CODE REVIEW

**Comment by richburdon** [comment]
Suggesting the following:

```ts
const test = () => {
  const client = new Client();

  const space1 = client.spaces.root; // Always present; cannot be changed (or shared).
  const space2 = client.spaces.query();
  const space3 = client.spaces.query({ type: 'ephemeral' });
  const space4 = client.spaces.create({ type: 'ephemeral' }); // Cannot be shared.

  console.log(space1.type);
};
```

Rationale:

- No special verbs (personal, ephemeral).
- Variants supported via `space.type` without type-specific API variants.
- Achieves default (now "root") space, but doesn't make this changeable; instead we augment spaces with features to reset/fork/transport objects.
- Normalize API on `client.spaces` noun.

**Comment by zhenyasav** [comment]
Counter proposal:


```ts

const client = new Client();

const defaultSpace            = client.spaces.default;
const nonEphemeralSpacesOnly  = client.spaces.query();
const includesEphemeralSpaces = client.spaces.query({ ephemeral: true });
const newEphemeralSpace       = client.spaces.create({ ephemeral: true }); // Cannot be shared.

```

- `root` is less intuitive than `default` in this feature, may imply more than we want to imply for now
- `type` opens up a cardinality in excess of what we need for now, and implies mutual exclusivity with any other types.

File: docs/docs/specs/v0.1/personal-space.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import { Client } from '@dxos/client';
+
+const client = new Client();
+
+await client.initialize();
+
+const identity = client.halo.identity.get(); // null by default
+
+// get the personal space, creating an identity is not required
+const space = client.getPersonalSpace();
```
Comment (dmaretskyi) [inline docs/docs/specs/v0.1/personal-space.md:22]: do we want to keep consistency with other getters and change this to `client.personalSpace`?

File: docs/docs/specs/v0.1/personal-space.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+// sense if a space is the personal space
+const isPersonalSpace = space.isPersonalSpace;
+
+// recreate the personal space and copy all data into the new space without history
+// (old space will disappear or archive on all devices)
+const newSpace = await client.createPersonalSpace();
```
Comment (dmaretskyi) [inline docs/docs/specs/v0.1/personal-space.md:29]: What happens when you join as a new device to an existing identity after creating the personal space?

File: docs/docs/specs/v0.1/personal-space.md (hunk 1 above)
Comment (zhenyasav) [reply docs/docs/specs/v0.1/personal-space.md:22]: yeah makes sense, we're naming this `default` or `root` space at the moment

**Comment by richburdon** [comment]
> Counter proposal:
> 
> ```ts
> const client = new Client();
> 
> const defaultSpace            = client.spaces.default;
> const nonEphemeralSpacesOnly  = client.spaces.query();
> const includesEphemeralSpaces = client.spaces.query({ ephemeral: true });
> const newEphemeralSpace       = client.spaces.create({ ephemeral: true }); // Cannot be shared.
> ```
> 
> * `root` is less intuitive than `default` in this feature, may imply more than we want to imply for now
> * `type` opens up a cardinality in excess of what we need for now, and implies mutual exclusivity with any other types.

- Re default: I think "default" doesn't convey the very special purpose of this space (i.e., always there, single player). `home`? I'm a bit torn with it being in the API at all since the current use case is very "onboarding" specific, but per our discussion I can see this evolving into being the "anchor" point for all of your spaces (the full graph expression). Also avoids isEphemeral` accessors. 

- Re type: I think we will represent this as a proto type enum and I'm generally more inclined to expose that to the low-level technical API. Keeping it as a property also makes queries more uniform/straight forward. In general trying to avoid special-casing the API. I just sense that this binary distinction is too specialized for the sake of syntax sugar?

However, feel free to move forward if you're not swayed by these arguments -- or let's finalize in our next design chat?

**Comment by zhenyasav** [comment]
- the default space is not "single player", it's shared across devices
- the type enum introduces cardinality that I don't think is intended, let's discuss

File: docs/docs/specs/v0.1/ephemeral-space.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+// TODO: how to name spaces for easy retrieval?
+
+// create an ephemeral space for the current window only
+const space = await client.createSpace({ ephemeral: true, replication: false });
+
+// sense if a space is ephemeral
+const isEphemeral = space.isEphemeral;
+
+// sense if replicating across windows
+const isReplicating = space.isReplicating;
```
Comment (wittjosiah) [inline docs/docs/specs/v0.1/ephemeral-space.md:36]: Is there any distinction between window replication and device replication?

**Pull request closed without merging**